### PR TITLE
Build docker images with GitHub Actions

### PR DESCRIPTION
**Pull request by sandhose**
This is heavily inspired by the [example in the docker/build-push-action](https://github.com/docker/build-push-action#complete-workflow) repo.

It builds two images: `zazuko/cube-creator-{api,app}`. Tags as follow: 
 - `sha-{commit}` on every commit
 - `edge` for the latest commit on master
 - `[branch]` for other branches
 - on git tags like `vX.Y.Z`, tags `vX`, `vX.Y`, `vX.Y.Z` and latest
 - `pr-X` on pull requests (builds but does not push)

### CODE REVIEW

**Comment by tpluscode** [comment]
Would be nice to also have a docker build on Pull Requests?

**Comment by sandhose** [comment]
Its the case, just that it is not pushed. Secrets are not available when building PRs, so we can't push them to Docker Hub. Also, it runs and push for each branches anyways

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
+name: docker
+
+on:
+  push:
+    branches:
+      - "**"
+    tags:
+      - "v*.*.*"
```
Comment (tpluscode) [inline .github/workflows/docker.yaml:8]: Aha, this makes me realise. My default choice would be to have separate version history for the API and UI. They would produce their own tags like `@cube-creator/core-api@v0.1.0`.

Unless you would like a single version for the entire repository @martinmaillard ?

Comment (martinmaillard) [reply .github/workflows/docker.yaml:8]: I would personally have opted for a single version for the entire repository, as it makes maintenance easier. Any argument for having separate versions?

Comment (tpluscode) [reply .github/workflows/docker.yaml:8]: Building and deploying separately. Many changes in the UI will be unrelated to the API - no version bump. And vice versa.

I'm not sure that a single version is going to make reduce maintenance

Comment (martinmaillard) [reply .github/workflows/docker.yaml:8]: Building and deploying will be automated so I don't see a problem deploying everything for every version, even if one of the parts didn't change.

Comment (vhf) [reply .github/workflows/docker.yaml:8]: @tpluscode ok for you if we go with a single version?

Comment (tpluscode) [reply .github/workflows/docker.yaml:8]: Yes, tentatively. I would like to see a proposal for managing this version. As far as tooling is concerned there is nothing that I know of which will help us.

To explain, we have a monorepo which is at least the UI and API which we can evolve somewhat independently by keeping their change history and versions. Tooling exists to manage this (`lerna version` or `atlassian changesets`)

The alternative is what? To ignore the `version` in packages and only keep the one in root `package.json`?

**Review by tpluscode — APPROVED** [review]

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (vhf) [reply .github/workflows/docker.yaml:8]: Unless we publish the API and the UI as packages I'm not sure it really matters. What's important at this point is having docker images to deploy

Comment (tpluscode) [reply .github/workflows/docker.yaml:8]: We package them as two docker images which is close enough, isn't it?

Comment (vhf) [reply .github/workflows/docker.yaml:8]: Right, I missed that part. Let's go with this and we'll revisit when we need to.

**Pull request merged**